boot/bootutil/src/loader.c

```diff
@@ -1933,3 +1939,189 @@ split_go(int loader_slot, int split_slot, void **entry)
     free(sectors);
     return rc;
 }
+
+#else /* MCUBOOT_DIRECT_XIP */
+
+/**
+ * Iterates over all slots and determines which contain a firmware image.
+ *
+ * @param state          Boot loader status information.
+ * @param slot_usage     A pointer to a zero-initialized array, which aim is to
+ *                       carry information about slots that contain an image.
+ *                       After return the corresponding array elements are set
+ *                       to a non-zero value if the given slots are in use
+ *                       (contain a firmware image).
+ * @param slot_cnt       The number of slots, which can contain firmware images.
+ *
+ * @return               The number of found images.
+ */
+static uint32_t
+boot_get_slot_usage(struct boot_loader_state *state, uint8_t *slot_usage,
```

Comment:
`slot_usage` is array, so you may directly declare it as array (`uint8_t slot_usage[]`). It is pointer to array anyway, just seems more intuitive, at least to me.